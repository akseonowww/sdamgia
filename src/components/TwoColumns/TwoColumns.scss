@import '../../variables';
@import '../../breakpoints';

.TwoColumns {
	&-Grid {
		grid-row-gap: 0;
	}

	&-Col {
		grid-column: 1/21;

		@include media-breakpoint-up(md) {
			grid-column: span 10;
		}

		&_1 {
			margin-bottom: 1.75rem;

			@include media-breakpoint-up(md) {
				margin-bottom: 0;
				padding: 0 0.4rem 1.5rem 0;

				border-right: 0.15rem solid $primary;
			}
		}

		&_2 {
			border-top: 0.15rem solid $primary;

			@include media-breakpoint-up(md) {
				border-top: none;
			}
		}
	}
}
