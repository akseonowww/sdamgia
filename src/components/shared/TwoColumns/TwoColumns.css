.TwoColumns-Grid {
	grid-row-gap: 0;
}

.TwoColumns-Col {
	grid-column: 1/21;
}

.TwoColumns-Col_1 {
	margin-bottom: 1.75rem;
}

.TwoColumns-Col_2 {
	border-top: 0.15rem solid #006;
}

@media (min-width: 768px) {
	.TwoColumns-Col {
		grid-column: span 10;
	}

	.TwoColumns-Col_1 {
		margin-bottom: 0;
		padding: 0 0.4rem 1.5rem 0;

		border-right: 0.15rem solid #006;
	}

	.TwoColumns-Col_2 {
		border-top: none;
	}
}
