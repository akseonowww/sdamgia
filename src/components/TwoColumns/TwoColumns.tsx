import React, { ReactNode } from 'react'

import Section from '../Section'
import Grid from '../Grid'
import './TwoColumns.scss'

interface ITwoColumnsProps {
  children: ReactNode[]
}

const TwoColumns = ({ children }: ITwoColumnsProps) => (
  <Section>
    <Grid className="TwoColumns-Grid">
      <div className="TwoColumns-Col TwoColumns-Col_1">{children[0]}</div>
      <div className="TwoColumns-Col TwoColumns-Col_2">{children[1]}</div>
    </Grid>
  </Section>
)

export default TwoColumns
