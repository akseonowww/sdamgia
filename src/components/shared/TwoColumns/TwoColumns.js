import React from 'react';

import './TwoColumns.css';

const TwoColumns = () => (
    <div>Две колонки</div>
);

export default TwoColumns;