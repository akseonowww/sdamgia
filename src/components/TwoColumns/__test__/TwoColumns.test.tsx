import React from 'react'
import { unmountComponentAtNode, render } from 'react-dom'
import { act } from 'react-dom/test-utils'
import pretty from 'pretty'

import TwoColumns from '../TwoColumns'

describe('TwoColumns', () => {
  let container: HTMLElement | null = null

  beforeEach(() => {
    container = document.createElement('div')
    document.body.appendChild(container)
  })

  afterEach(() => {
    if (container) {
      unmountComponentAtNode(container)
      container.remove()
      container = null
    }
  })

  it('renders without crashing', () => {
    act(() => {
      render(
        <TwoColumns>
          <div>First Col</div>
          <div>Second Col</div>
        </TwoColumns>,
        container
      )
    })
    expect(container && container.textContent).toBe('First ColSecond Col')
  })

  it('should render a content', () => {
    act(() => {
      render(
        <TwoColumns>
          <div>First Col</div>
          <div>Second Col</div>
        </TwoColumns>,
        container
      )
    })
    if (container) expect(pretty(container.innerHTML)).toMatchSnapshot()
  })
})
