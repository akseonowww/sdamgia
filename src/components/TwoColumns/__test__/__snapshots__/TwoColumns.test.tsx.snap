// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TwoColumns should render a content 1`] = `
"<section class=\\"Section\\">
  <div class=\\"Grid TwoColumns-Grid\\">
    <div class=\\"TwoColumns-Col TwoColumns-Col_1\\">
      <div>First Col</div>
    </div>
    <div class=\\"TwoColumns-Col TwoColumns-Col_2\\">
      <div>Second Col</div>
    </div>
  </div>
</section>"
`;
